connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -365,6 +372,16 @@ class KustoClient(val engineKcsb: ConnectionStringBuilder, val ingestKcsb: Conne
 
     shouldIngest
   }
+
+  def executeEngine(database: String, command: String, crp: ClientRequestProperties, retryConfig: Option[RetryConfig] = None): KustoOperationResult = {
+    KDSU.retryFunction(() => engineClient.execute(database, command, crp), retryConfig.getOrElse(this.retryConfig),
```

Comment:
I find this (existing) design a bit strange. `engineClient`, `dmClient` and `ingestClient` are all basically singletons per KCSB that live as members within `KustoClient` - a class that I'd expect those 3 to be an **instance** of.

I'm sure there was wisdom to it beyond what I see in snippets here, and even if not, there isn't really anything to change about it here. So just calling it out, and asking if you know of wisdom behind it.

Reply:
Decided on renaming 
With a suggestion for the future to split into three wrapper classes and a holder class